web/src/types.ts

```diff
@@ -6,13 +6,67 @@ export type MatchedMessage = {
 
 // TODO/typescript: Move this to message_store
 export type MessageType = "private" | "stream";
+export type MessageContentType = "text/html" | "text/x-markdown";
+export type MessageReactionType = "unicode_emoji" | "realm_emoji" | "zulip_extra_emoji";
+
+// TODO/typescript: Move these types to message_store
+
+export type DisplayRecipient = {
+    email: string;
+    full_name: string;
+    id: number;
+    is_mirror_dummy: boolean;
+};
+
+export type MessageEditHistory = {
+    user_id: number | null;
+    timestamp: number;
+    prev_content: string;
+    prev_rendered_content_version: number;
+    prev_stream: number;
+    prev_topic: string;
+    stream: number;
+    topic: string;
+};
+
+export type MessageReaction = {
+    emoji_name: string;
+    emoji_code: string;
+    reaction_type: MessageReactionType;
+    user_id: number;
+    user: DisplayRecipient;
+};
+
+// TODO/typescript: Move this to server_events
+export type TopicLink = {
+    text: string;
+    url: string;
+};
 
 // TODO/typescript: Move this to message_store
 export type RawMessage = {
+    avatar_url: string;
+    client: string;
+    content: MessageContentType;
+    content_type: string;
+    display_recipient: string | DisplayRecipient[];
```

Comment:
I think probably it make more sense for the `DisplayRecipient` type to be this, and the typed defined above to be called `DisplayRecipientUser`.